packages/payments-client/src/components/pages/payment.tsx

```diff
@@ -35,10 +40,13 @@ export const PaymentsPage: React.FC = () => {
     setLoading(true)
   }, [setMerchantKey])
 
-  if (loading) {
+  if (!data || loading) {
```

Comment:
Cam we move this logic below the H3 so the H3 is always rendered and only the form is hidden when loading